### ISSUE TITLE
1.0b145 unpickling error

### ISSUE THREAD

**Issue by colinallen** [inpho/topic-explorer#177]
Notebook fails at first line.

Running from notebook, using serial load function.
---------------------------------------------------------------------------
UnpicklingError                           Traceback (most recent call last)
<ipython-input-1-05dd64fba2b2> in <module>()
----> 1 from corpus import *
      2 from vsm import *

/Users/colin/corpora/Chinese/Handian/notebooks/corpus.py in <module>()
     19     c = CorpusSent.load(config.get('main', 'corpus_file'))
     20 else:
---> 21     c = Corpus.load(config.get('main', 'corpus_file'))
     22 context_type = config.get('main', 'context_type')
     23 ctx_metadata = c.view_metadata(context_type)

/Users/colin/anaconda/lib/python2.7/site-packages/vsm/zipfile.pyc in wrapper(*args, **kwargs)
     27     def wrapper(*args, **kwargs):
     28         inject_czipfile()
---> 29         value = f(*args, **kwargs)
     30         inject_zipfile()
     31         return value

/Users/colin/anaconda/lib/python2.7/site-packages/vsm/corpus/base.pyc in load(file, corpus_dir, corpus_file, words_file, metadata_file)
    661             c = Corpus([], remove_empty=False)
    662             c.corpus = arrays_in['corpus']
--> 663             c.words = arrays_in['words']
    664             c.context_types = arrays_in['context_types'].tolist()
    665             try:

/Users/colin/anaconda/lib/python2.7/site-packages/numpy/lib/npyio.pyc in __getitem__(self, key)
    222                 return format.read_array(bytes,
    223                                          allow_pickle=self.allow_pickle,
--> 224                                          pickle_kwargs=self.pickle_kwargs)
    225             else:
    226                 return self.zip.read(key)

/Users/colin/anaconda/lib/python2.7/site-packages/numpy/lib/format.pyc in read_array(fp, allow_pickle, pickle_kwargs)
    636             pickle_kwargs = {}
    637         try:
--> 638             array = pickle.load(fp, **pickle_kwargs)
    639         except UnicodeError as err:
    640             if sys.version_info[0] >= 3:

UnpicklingError: state is not a dictionary

**Comment by JaimieMurdock** [comment]
I was able to replicate this issue. The only major change in the vsm code was to shift the data type of the c.words array from `np.unicode_` to `np.object_`, but that could be introducing an error to our serialization system. Investigating now.

**Comment by JaimieMurdock** [comment]
The error was in `vsm.zipfile` which injects the library `czipfile` for quicker loading of data. Funnily enough, Googling for "jupyter czipfile" leads to our own bug report: https://github.com/inpho/vsm/issues/140

I disabled the zipfile injection manually in `vsm/zipfile.py`:

```python

def use_czipfile(f):
    def wrapper(*args, **kwargs):
        #inject_czipfile()
        value = f(*args, **kwargs)
        #inject_zipfile()
        return value

    return wrapper
```

Now looking into a fix for the `if/elif` branch in `vsm/corpus/base.py` that will properly disable this temporarily by not calling `use_czipfile` on the load functions when in notebook mode.

**Comment by JaimieMurdock** [comment]
Fixed in `vsm 0.4b7`, completing builds for all platforms then will close issue with update to `topicexplorer 1.0b146`
